lib/postgrex/extensions/polygon.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Postgrex.Extensions.Polygon do
+  @moduledoc false
+  import Postgrex.BinaryUtils, warn: false
+  use Postgrex.BinaryExtension, send: "poly_send"
+  alias Postgrex.Extensions.Polygon
+  alias Postgrex.Extensions.Point
+
+  def encode(_) do
+    quote location: :keep do
+      %Postgrex.Polygon{vertices: vertices} when is_list(vertices) ->
+        len = <<length(vertices)::int32>>
+        vert = Enum.map(vertices, &Point.encode_point(&1, Postgrex.Polygon))
```

Comment:
Please use reduce to build iodata and append with cons: `[acc | point]` as no stack required. Please do elsewhere